src/cmdlib.sh

```diff
@@ -628,6 +628,41 @@ strip_out_lockfile_digests() {
     mv "$1.tmp" "$1"
 }
 
+# Strip out the arch field from lockfiles to make them archless.
+strip_out_lockfile_arches() {
+    python3 -c '
+import sys, json
+lockfile = json.load(sys.stdin)
+new_packages = {}
+for pkg, meta in lockfile["packages"].items():
+    evra = meta.get("evra")
+    if evra is None or evra.endswith(".noarch"):
+        new_packages[pkg] = meta
+        continue
+    new_packages[pkg] = {
+        "evr": evra[:evra.rindex(".")]
+    }
+lockfile["packages"] = new_packages
+json.dump(lockfile, sys.stdout)
+' < "$1" > "$1.tmp"
+    mv "$1.tmp" "$1"
+}
+
+# Create the autolock for a given version
+generate_autolock() {
+    local version=$1; shift
+    local autolockfile="${tmprepo}/tmp/manifest-autolock-${version}.json"
+    if [ ! -f "${autolockfile}" ]; then
```

Comment:
In what cases would the `$autolockfile` already exist? should we optimize for that rather than just overwrite it?

Reply:
Hmm, I don't follow. We're not overwriting it here.

Reply:
what I was trying to say is that we should just overwrite unconditionally and asking why we shouldn't.

Reply:
We could, but it's pretty safe to not I think. The case is hit because both `cosa fetch` and `cosa build` need it. The result is mechanically derived and named per version, so this function is following a "cache miss" pattern where it populates it when missing.